examples/GroveButton/GroveButton.ino

```diff
@@ -17,23 +17,37 @@ Licensed under the MIT license. See LICENSE file in the project root for full li
 // upm - Version: Latest 
 #include "upm_utilities.h"
 
+//Set to false if SSH and set to true if Serial Connection
+#define IS_SERIAL_OR_SSH true
 //A 512 offset is required for sub-platforms.  516 corresponds to digital pin 4, or D4.
 #define BUTTON_PIN 516
 
+void print(char *msg)
+{
+	#ifdef IS_SERIAL_OR_SSH
+		DebugSerial.println(msg);
+	#else
+		printf("%s\n", msg);
+	#endif
+}
```

Comment:
Looks good. One thing I can suggest is that take second argument as %s. e.g.
print(char *msg, char *format) and pass "%s\n" as format to printf. It will be ignored for debugSerial.println. That way if someone wanted to pass %d and print out some integers, this API will still work.

Reply:
Good idea. But I wonder how that will work.

String Example:
print("released", "%s\n");

Integer Example:
print(5, "%d\n");

In the string example, the type of the first argument is string, and in the Integer example the first argument is integer. There may be other ways to get this working, but probably not worth the efforts.

May be we can merge this now, and create a separate pull request with those changes?

Reply:
This is called function overloading in C++. https://www.tutorialspoint.com/cplusplus/cpp_overloading.htm
 I think once you do the change in all samples we might need it sooner rather than later. Its your call.

Reply:
@sasmita  Sorry I don't have bandwidth anymore to focus on this issue. 
Seems like you are confident that the merge should be made? If that is the case I will go ahead and merge the PR. Since you also work in arduino create, you should be able to address if anything comes out of it. Please let me know if I should make the merge.

Reply:
I agree we don't have to spend so much time on this one. Please merge it.